compiler/rustc_resolve/src/rustdoc.rs

```diff
@@ -410,8 +410,10 @@ fn parse_links<'md>(doc: &'md str) -> Vec<Box<str>> {
     while let Some(event) = event_iter.next() {
         match event {
             Event::Start(Tag::Link(link_type, dest, _)) if may_be_doc_link(link_type) => {
-                if let Some(display_text) = collect_link_data(&mut event_iter) {
-                    links.push(display_text);
+                if matches!(link_type, LinkType::Inline | LinkType::ReferenceUnknown | LinkType::Reference) {
+                    if let Some(display_text) = collect_link_data(&mut event_iter) {
+                        links.push(display_text);
+                    }
```

Comment:
i think the panic happens because you also need to consider `Shortcut` and `ShortcutUnknown`: https://docs.rs/pulldown-cmark/latest/pulldown_cmark/enum.LinkType.html#variant.Shortcut

Reply:
After adding `Shortcut` and `ShortcutUnknown`, the panic still happens.